app/lib/features/segment_deletion_config.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+module Features
+  class SegmentDeletionConfig
+    def self.configure(config = {})
+      @configuration = new(config)
+    end
+
+    def self.configuration
+      (@configuration || configure)
+    end
+
+    def self.config
+      configuration.config
+    end
+
+    def self.enabled?
+      configuration.enabled?
+    end
+
+    def initialize(config = {})
+      format_config(config)
+      check_validity(config)
+    end
+
+    def self.valid_environment?(environment: Rails.env)
+      # CAUTION!:
+      # This is because this feature destroys real data unless the call to segment is stubbed and without real credentials in the config (test environment),
+      # so we prevent this feature to be used in preview or development thinking that be just checking around while actually destroying real data by accident
+      %w[production test].include?(environment)
```

Comment:
I'd say we should be able to test it locally and in preview. Probably ask/read doc in Segment @Martouta. How to isolate users depending on the sources that they can manage.

https://segment.com/docs/guides/account-management/i-want-to-set-up-segment-for-multiple-environments-development-production-testing-etc-should-i-create-multiple-workspaces-or-multiple-sources/

The only issue is that the GDPR authentication is using email / password to get a token. So it is scoped by user
Solution would probably be Segment providing a way to restrict users permission

Reply:
Look in the [documentation of this API](https://segment.com/docs/guides/best-practices/user-deletion-and-suppression). There is this sentence:

> Regulations enable you to issue a single request to delete and suppress data about a user by userId. All regulations are scoped to your workspace and target all sources within the workspace

In other words: whatever we do with this API, affects all sources 😱 
I will ask them if there is anything we can do about it 🤔

Reply:
Suggestion by Hery which I will try:

> Do you think it is possible per source permission?
> Try adding a user for system-dev and not system-pre
> and use the GDPR API